handler/src/main/java/io/netty/handler/timeout/WriteTimeoutHandler.java

```diff
@@ -70,6 +70,8 @@
 
     private final long timeoutNanos;
 
+    private Set<ScheduledFuture<?>> timeouts;
```

Comment:
I wonder if we need to keep a collection of timeouts. Because the time is fixed, each new event will alway fire later (in time) than the previous event. Can we just keep a reference to the "last" event, and cancel it when necessary (scheduling new event, cleanup, etc..)?